### PR TITLE
Add Space To Batch and its opposite.

### PR DESCRIPTION
**Pull request by ebarsoum**
Add SpaceToBatch, BatchToSpace and Tile OP.

### CODE REVIEW

**Comment by dzhulgakov** [comment]
Maybe use generic op version that doesn't hard-code data layout (e.g. something like https://www.tensorflow.org/api_docs/cc/class/tensorflow/ops/space-to-batch-n-d#classtensorflow_1_1ops_1_1_space_to_batch_n_d). In general ONNX doesn't hard code NCHW format unless necessary.

Just curious, for which model do we need it? Or is it because of CoreML?

**Comment by ebarsoum** [comment]
@dzhulgakov  We can added the generic one in the feature, the reason I added this one is because it is already implemented in Caffe 2 and yes for CoreML compact.

File: docs/Operators.md [hunk 1]
```diff
@@ -2314,6 +2354,42 @@
 </dl>
 
 
+### <a name="SpaceToBatch"></a><a name="spacetobatch">**SpaceToBatch**</a>
+
+  SpaceToBatch for 4-D tensors<T>. Zero-pads and then rearranges (permutes) blocks 
+  of spatial data into batch. More specifically, this op outputs a copy of the input tensor where values 
+  from the height and width dimensions are moved to the batch dimension. After the zero-padding, both
```
Comment (dzhulgakov) [inline docs/Operators.md]: you're probably missing padding arguments...

**Comment by ebarsoum** [comment]
@dzhulgakov  Change it to SpaceToDepth and its opposite instead, which is what we want. I agree for your comment for the previous OP.

**Comment by ebarsoum** [comment]
@dzhulgakov  here the TF version:

https://www.tensorflow.org/api_docs/python/tf/depth_to_space

Why extend it require an extra attribute? Isn't just a description update?

**Comment by dzhulgakov** [comment]
you're right, if we're talking about square blocks it's just a description update

**Pull request merged**